pkg/gameservers/migration.go

```diff
@@ -46,35 +45,39 @@ import (
 // event happens on a node, and a Pod is recreated with a new Address for a
 // GameServer
 type MigrationController struct {
-	baseLogger       *logrus.Entry
-	podSynced        cache.InformerSynced
-	podLister        corelisterv1.PodLister
-	gameServerSynced cache.InformerSynced
-	gameServerGetter getterv1.GameServersGetter
-	gameServerLister listerv1.GameServerLister
-	nodeLister       corelisterv1.NodeLister
-	nodeSynced       cache.InformerSynced
-	workerqueue      *workerqueue.WorkerQueue
-	recorder         record.EventRecorder
+	baseLogger               *logrus.Entry
+	podSynced                cache.InformerSynced
+	podLister                corelisterv1.PodLister
+	gameServerSynced         cache.InformerSynced
+	gameServerGetter         getterv1.GameServersGetter
+	gameServerLister         listerv1.GameServerLister
+	nodeLister               corelisterv1.NodeLister
+	nodeSynced               cache.InformerSynced
+	workerqueue              *workerqueue.WorkerQueue
+	recorder                 record.EventRecorder
+	syncPodPortsToGameServer func(*agonesv1.GameServer, *corev1.Pod) error
```

Comment:
Similarly, this is always set to the nil function in the tests. Is this another place with missing test coverage?

Reply:
This is tested here: https://github.com/googleforgames/agones/blob/3c38876382d505244af68489211d8916dce07596/pkg/gameservers/gameservers_test.go#L107-L164

It's again slightly indirected from the hook value (these were both examples of how I tested the hook callers prior to the hooks getting factored out and maybe they could be better now?).